subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/DependencyTree.groovy

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.generator
+
+import org.gradle.internal.Pair
+
+class DependencyTree {
```

Comment:
I'm a bit lost at what this class does. Some class documentation about the problem it solves would help review it.

Also I'm curious: We talked about creating dependencies purely based on simple arithmetic like "class n depends on class n - 3", so I'm not quite sure why we need to hold state at all?